src/Microsoft.DocAsCode.Build.Common/ExtractSearchIndex.cs

```diff
@@ -1,45 +1,57 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-// todo : move to Plugin
 
-namespace Microsoft.DocAsCode.Build.Engine.ExtractSearchData
+namespace Microsoft.DocAsCode.Build.Common
 {
     using System;
     using System.Collections.Generic;
+    using System.IO;
     using System.Linq;
     using System.Text;
     using System.Text.RegularExpressions;
-    using System.IO;
-
-    using Newtonsoft.Json;
-    using HtmlAgilityPack;
+    using System.Composition;
+    using System.Collections.Immutable;
 
-    using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Plugins;
+    using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.MarkdownLite;
 
-    public class ExtractSearchIndexFromHtml
+    using HtmlAgilityPack;
+    using Newtonsoft.Json;
+
+    [Export(nameof(ExtractSearchIndex), typeof(IPostProcessor))]
+    public class ExtractSearchIndex:IPostProcessor
     {
+        private static readonly Regex RegexWhiteSpace = new Regex(@"\s+", RegexOptions.Compiled);
+
+        public string Name => nameof(ExtractSearchIndex);
         public const string IndexFileName = "index.json";
 
-        private static readonly Regex RegexWhiteSpace = new Regex(@"\s+", RegexOptions.Compiled);
+        public ImmutableDictionary<string, object> Update(ImmutableDictionary<string, object> metadata)
+        {
+            if (!metadata.ContainsKey("_enableSearch"))
+            {
+                metadata = metadata.Add("_enableSearch", true);
+            }
+            return metadata;
+        }
 
-        public static void GenerateFile(List<ManifestItem> manifest, string baseDir)
+        public Manifest Process(Manifest manifest, string baseDir)
```

Comment:
Seems no need to check null for plugin input, since we can guarantee the input is not null. But we should check null for the plugin output.